src/hooks/useAutoSlippageTolerance.ts

```diff
@@ -43,8 +45,19 @@ function guesstimateGas(trade: Trade<Currency, Currency, TradeType> | undefined)
         // V3 gas costs scale on initialized ticks being crossed, but we don't have that data here.
         // We bake in some tick crossings into the base 100k cost.
         gas += V3_SWAP_BASE_GAS_ESTIMATE + route.pools.length * V3_SWAP_HOP_GAS_ESTIMATE
+      } else if (route.protocol === Protocol.MIXED) {
+        const sections = partitionMixedRouteByProtocol(route as MixedRoute<Currency, Currency>)
+        let acc = 0
```

Comment:
`acc` is unnecssary - you should instead just have L53 and L55 use `gas +=` to be more explicit about what you're adding up.

You should also use `reduce` to make it more clear what is happening in the code:

    gas += sections.reduce((gas, section) => {
      if (section.every((pool) => pool instanceof Pool)) {
        return gas + V3_SWAP_BASE_GAS_ESTIMATE + section.length * V3_SWAP_HOP_GAS_ESTIMATE
      } else if (section.every((pool) => pool instanceof Pair)) {
        return gas + V2_SWAP_BASE_GAS_ESTIMATE + (section.length - 1) * V2_SWAP_HOP_GAS_ESTIMATE
      } else {
        console.warn('Invalid section')
        return gas
      }
    })

Reply:
I like the .reduce